# CREST evaluation

## Objectives:
In this investigation, the testing of taxonomic classification software with different databases is carried out because a reliable assessment of taxonomy at species level is indispensable nowadays for understanding microbiomes. 

This project was carried out in order to investigate the performance of the CREST against different classifiers. With this aim in mind, the objectives are to:

* Test and evaluate the performance of each software after a removal-of-taxa cross-validation test.
* Test and evaluate the performance of each software after a ten-fold cross-validation test.
------------

## Software tested:
* LCA classifier (CREST)
* SINTAX

-----------
## Databases tested:
### UNITE:
* UNITE v7 (23,264 seq) implemented in CREST (Lanzen et al.,2012)
* UNITE (53,693 seq) formatted for SINTAX http://www.drive5.com/usearch/manual/sintax_downloads.html and filtered to contained the same accession numbers as the version of CREST.

## MATERIALS AND METHODS

Two types of test were experienced: removal-of-taxa cross-validation and ten-fold cross-validation.

For each test, the software was first tested by classifying full length sequences and on sequences that had been shortened by an Illumina MiSeq v3 (MSv3) simulator to 250bp and that inserted errors characteristic of this sequencing platform (Huang et al., 2012).


The removal-of-taxa cross-validation consisted of randomly removing a whole taxon from the database and to classify the removed sequences using the truncated database generated that does not contain the taxon itself.

*This means for example that if the class Agaricomycetes is selected to be removed, then all the reference sequences that belongs to this taxon and its child taxa are extracted and deleted from the database. The extracted sequences are then classified against the truncated database. We based this test on the removal of whole classes, orders, families and genera.*

The ten-fold cross-validation on the other hand, consisted of extracting 10% of sequences picked randomly from the database and to classify them using the truncated database.



### CREST - LCA Classifier - Evaluation:
#### Implementation:
The LCA Classifier program of CREST uses alignment data generated by an initial alignment of the dataset against a reference database using BLASTN. The classification is then carried out by the program based on a subset of best matching alignments to the Lower Common Ancestor (LCA) of this subset. The program uses minimum similarity filters for the classification of 16S rRNA but in the case of the ITS classification, these filters were disabled (Lanzen et al.,2012).

### Software evaluation

####Removal-of-taxa cross validation
##### Scripts used:
* Scripts/Bash/pipeline_cpx_lca_FL.sh
* Scripts/Bash/pipeline_cpx_lca_250bp.sh
* Scripts/Python/cpx_lca.py (*Implemented in CREST*)
* Scripts/Python/evaluate_lca.py (*Implemented in CREST*)

##### Pipeline:

For this test, we made two distinct pipelines, one to classify full length sequences (*pipeline cpx_lca_FL.sh*) and one for the shortened sequences (*pipeline_cpx_lca_250bp.sh*).


The creation of the test datasets and truncated reference databases is the first step. To do that the Python script "cpx_lca.py" was implemented. This script takes the UNITE database as argument and a list of taxon names (Tedersoo et al., 2014) of the rank that we want to extract. Then it selects randomly a taxon from the list and extracts all the sequences corresponding to this taxon and their children from the reference truncated database. The process is illustrated in Figure 1.

![](/home/amelie/flow.png)

**Figure 1. Global functioning of the creation of new datasets.** *The original UNITE database is divided in two output, one containing the isolated sequences we want to classify and the truncated database that does not contains those sequence anymore.*

An optional step is added after the creation of the dataset to simulate 250bp read from Illumina MSv3 sequencing by using ART-Illumina.


To classify the 250bp and the full length sequences, the next step was the indexing of the new truncated database for BLASTN. Then the alignment of the sequences against the new database is done with BLASTN and the "-task megablast" option.

The Python script written for the analysis of the output generated by CREST (*evaluate_lca.py*) takes the classification file resulting from the LCA Classifier of CREST and calculates the number of true positives, true negatives, false positives and false negatives at each rank ranging from domain to genus. These results can be defined by the correspondence of the taxa from the classified sequences against how they were supposed to be classified in the reference database (Figure 2).

![](/home/amelie/score.png)
**Figure 2. Illustration of the evaluation of the classification.** *The evaluation is counting the
number of true positives, true negatives, false positives and false negatives. The top set of circle
represent the taxonomy delivered by the software after the test while the bottom set represent the
original taxonomy as found on the database. If the two strand have the same color, it means that
they belongs to the same taxon. If there is nothing then the sequence is not classified to this level,
the taxon is unknown.*


With these results, The accuracy of classification at each rank is calculated using the following equation:

(1) $$Accuracy = (TP+TN)/(TP+TN+FP+FN)$$ 

For the removal-of-taxa cross-validation test the sequences that were submitted for a classification belonged to a taxon that was removed from the database. Therefore no classification from the taxon removed to the lowest level is possible and the result is counted as a true negative.

#### Ten-fold cross-validation
##### Scripts used:
* Scripts/Python/ten_fold_lca.py
* Scripts/Bash/ten_fold_pipeline_lca_250bp.sh
* Scripts/Bash/ten_fold_pipeline_lca_FL.sh
* Scripts/Python/evaluate_lca.py (*Implemented in CREST*)

#####Pipeline
The script written for this test, "ten_fold_lca.py", extracts randomly 10 percent of all the sequence in the UNITE database and output a new database containing all the sequences from UNITE except those extracted.
*This script is easily editable to change the percentage of sequences extracted*

The scripts for the overall pipeline of full length sequences classification and 250bp
sequences classification are respectively called "ten_fold_pipeline_lca_FL.sh"
and "ten_fold_pipeline_lca_250bp.sh" and the evaluation is done with the same
"evaluate_lca.py" script and the accuracy calculated with the equation (1).


### For Sintax:
The SINTAX evaluation follow the same pipeline as CREST but a cut off of 0.6 was chosen.

#### Implementation
The SINTAX algorithm predicts taxonomy by using k-mer similarity to identify
top hit in a reference database and provides bootstrap confidence for all ranks in
the prediction. (Edgar, 2016)

####
**Scripts used:**

	* Scripts/Python/cpx_sintax.py


------------


### For Sintax:
**Scripts used:**

	* Scripts/Python/random10sintax.py

---------
The duration of each test is calculated with the `user time` value given by the command `time` of bash. In the case of the ten-fold cross-validation, the time is calculated for the whole pipeline and for the removal-of-taxa cross-validation the time is a sum of the execution time of the pipeline for each taxonomic level extracted.


